shark_turbine/kernel/wave/register_analysis.py

```diff
@@ -33,16 +42,43 @@ def set_register_shape(trace: CapturedTrace, custom: CustomOp) -> None:
                     idx
                 ]
             )
-            set_register_shape(trace, iter_arg)
+            set_register_shape(trace, iter_arg, vector_map)
             custom.fx_node.thread_shape = iter_arg.fx_node.thread_shape
             break
+        elif isinstance(custom_user, ReduceOp):
+            get_thread_shape = lambda reg_shape: max(reg_shape)
+            # Check that dim is non-reduction && in hw_constraint.vector_shape.
+            is_valid_dim = lambda dim: dim != custom_user.dim and dim in vector_map
+            # TODO: Modify num_reduction_dims once we add support for multi-dim reduction.
+            num_reduction_dims = 1
```

Comment:
Can you query this from the ReduceOp and assert if the number of reduction dims is greater than 1?

Reply:
Done!

Reply:
thanks!